6238_prototype/src/main/java/frc/robot/commands/Intake/RetractIntakeCommand.java

```diff
@@ -0,0 +1,31 @@
+package frc.robot.commands.Intake;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.BallSubsystem;
+
+public class RetractIntakeCommand extends CommandBase{
+    private final long timeLimit;
+    BallSubsystem ball;
+    long startTime;
+    public RetractIntakeCommand(BallSubsystem ball) {
+        this.ball = ball;
+        addRequirements(ball);
+        timeLimit = 2000;
+        startTime = System.currentTimeMillis();
+    }
+
+    @Override
+    public void execute() {
+        ball.retract();
+        double bottomMotorSpeed = 0.8;
+        double upperMotorSpeed = -0.1;
```

Comment:
Don't need local variable  `bottomMotorSpeed` and `upperMotorSpeed` any more since you are just calling `setSpeed` on the next line.

That said, the constants `0.8` and `-0.1` should be constants shared with `RunIntakeCommand` to avoid code duplication.

The upperMotorSpeed is now defined in terms of RPM since we implemented PID control for the upper motor.  That means this code should have the `upperMotorSpeed` be something like `-100` so that the shooter rollers are rotating slowly inward.